File: commands/music/stop.js
/* global guild controller */

module.exports = {
  name: 'stop',
  description: 'Stannar botens nuvarande musik',
  aliases: [ 'stanna', 'st' ],
  group: 'music',
  usage: 'stop',
  example: 'stop',
  serverOnly: true,
  adminOnly: false,
  execute( message, args ) {
    if ( guild.voiceConnection ) {
      controller.stop();

      return message.reply( 'stoppar boten' ).then( msg => { msg.delete( 10000 ); } );
    } else return message.reply( 'kan inte stanna boten då boten inte spelar något' ).then( msg => { msg.delete( 10000 ); } );
  },
};

File: commands/music/stop.go
package music

import (
	"github.com/digitalungdom-se/dub/internal"
	"github.com/digitalungdom-se/dub/pkg"
)

var Stop = internal.Command{
	Name:        "stop",
	Description: "Stannar botens nuvarande musik",
	Aliases:     []string{"stanna", "st"},
	Group:       "music",
	Usage:       "stop",
	Example:     "stop",
	ServerOnly:  true,
	AdminOnly:   false,

	Execute: func(ctx *pkg.Context, server *internal.Server) error {
		ctx.Delete()
		server.Controller.Stop()

		return nil
	},
}
